payments.md

```diff
@@ -110,16 +47,173 @@ type SpendVoucher struct {
     // Lane specifies which 'lane' of the payment channel this voucher is for.
     // Lanes may be either open or closed, a voucher for a closed lane may not be redeemed
     Lane uint64
-    
-    // Merges specifies a list of lane-nonce pairs that this voucher will close. 
+
+    // Merges specifies a list of lane-nonce pairs that this voucher will close.
     // This voucher may not be redeemed if any of the lanes specified here are already
-    // closed, or their nonce specified here is lower than the nonce on-chain.
-    Merges []Pair<uint64, uint64>
+    // closed, or their nonce specified here is lower than the nonce of the lane on-chain.
+    Merges []MergePair
+
+    TimeLock uint64
+
+    SecretPreimage []byte
+
+    RequiredSector []byte
+  
+    DataCommitment []byte
+
+    MinCloseHeight uint64
     
     Sig Signature
 }
+
+type MergePair struct {
+    Lane uint64
+    Nonce uint64
+}
 ```
 
+```go
+type PaymentChannel struct {
+    From Address
+    To Address
+
+    ChannelTotal TokenAmount
+    ToSend TokenAmount
+
+    ClosingAt uint64
+    MinCloseHeight uint64
+
+    LaneStates map[uint64]LaneState
+}
+
+type LaneState struct {
+    Nonce uint64
+    Redeemed TokenAmount
+}
+
+func (paych *PaymentChannel) validateSignature(sv SpendVoucher) {
+    if msg.From == paych.From {
+        ValidateSignature(sv.SerializeNoSig(), sv.Signature, paych.To)
+    } else if msg.From == paych.To {
+        ValidateSignature(sv.SerializeNoSig(), sv.Signature, paych.From)
+    } else {
+        Fatal("bad programmer")
+    }
+}
+
+func (paych *PaymentChannel) UpdateChannelState(sv SpendVoucher, secret []byte, pip *PieceInclusionProof) {
```

Comment:
I am not sure I like the idea of hard-coding the `pip` here. In fact, payment channels should be completely agnostic from the rest of filecoin.

In other words, we could rename this payment channel actor the retrieval market payment channel, but that would be a bad idea.

Proposal: define the method of another contract that should be called and return true in order to unclock the funds, in other words, making fund unlocking before timeout to be contingent on another contract execution.